src/Apollo/NcrReactionsProjector.php

```diff
@@ -0,0 +1,181 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Apollo;
+
+use Aws\DynamoDb\DynamoDbClient;
+use Gdbots\Ncr\Aggregate;
+use Gdbots\Ncr\Event\NodeProjectedEvent;
+use Gdbots\Ncr\Exception\NodeNotFound;
+use Gdbots\Ncr\Ncr;
+use Gdbots\Ncr\Repository\DynamoDb\NodeTable;
+use Gdbots\Ncr\Repository\DynamoDb\TableManager;
+use Gdbots\Pbj\Message;
+use Gdbots\Pbj\MessageResolver;
+use Gdbots\Pbj\WellKnown\Microtime;
+use Gdbots\Pbj\WellKnown\NodeRef;
+use Gdbots\Pbjx\DependencyInjection\PbjxProjector;
+use Gdbots\Pbjx\Event\PbjxEvent;
+use Gdbots\Pbjx\EventSubscriber;
+use Gdbots\Pbjx\Pbjx;
+use Gdbots\Schemas\Ncr\Enum\NodeStatus;
+
+class NcrReactionsProjector implements EventSubscriber, PbjxProjector
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'triniti:apollo:mixin:has-reactions.projected'  => 'onNodeProjected',
+            'triniti:apollo:event:reactions-added'          => 'onReactionsAdded',
+        ];
+    }
+
+    public function __construct(
+        protected DynamoDbClient $client,
+        protected TableManager $tableManager,
+        protected Ncr $ncr,
+        protected bool $enabled = true
+    ) {
+    }
+
+    public function onNodeProjected(NodeProjectedEvent $pbjxEvent): void
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        $node = $pbjxEvent->getNode();
+        $nodeRef = $node->generateNodeRef();
+        $lastEvent = $pbjxEvent->getLastEvent();
+
+        if (NodeStatus::DELETED->value === $node->fget('status')) {
+        $context = ['causator' => $lastEvent];
+        $reactionsRef = $this->createReactionsRef($nodeRef);
+        $this->ncr->deleteNode($reactionsRef, $context);
+        return;
+    }
+
+        $reactions = $this->getReactions($nodeRef, $lastEvent) ?? $this->createReactions($nodeRef);
+
+        $this->mergeNode($node, $reactions);
+        $this->projectNode($reactions, $lastEvent, $pbjxEvent::getPbjx());
```

Comment:
@bhsiao55 projectNode problematic here as it will overwrite the whole node whenever a target is projected. (i.e. an article with reactions).  this should check if the node exists, if not create it.  can also be setup on the ".published" or ".created" instead of ".projected". which would ensure the reactions node is created one time and never overwritten later.

Reply:
@gdbrown I was using the same logic that article stats was using.  For example when an article title is updated the reactions node title field is updated as well.  You thinking we don't need this type of update for the reactions node?

Reply:
@bhsiao55 can't use that strategy with atomic counters.  this new strategy requires we only create teh node once.  the title, updated at, etag, etc.  is not important to keep in sync on the reactions node.